catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/utils/DorisUtils.java

```diff
@@ -65,4 +76,40 @@ public static Map<String, String> extractPropertiesFromSql(String createTableSql
     }
     return properties;
   }
+
+  public static Optional<Transform> extractPartitionInfoFromSql(String createTableSql) {
+    try {
+      String[] lines = createTableSql.split("\n");
+      for (String line : lines) {
+        if (line.contains(PARTITION_BY)) {
```

Comment:
Since the syntax of the partition by statement is fixed, we can use a regular here to make the code clear

Reply:
done